content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __DefaultValue__ | Optional | Allows you to provide the default value for the property | `DefaultValue="1000.0"`|
+| __DisplayGroup__ | Optional | Allows slots to be grouped visually | `DisplayGroup="Tests"` |
+----
+
+### Return
+
+__Return__ is used by the [Input](#input) element, represents and output data slots on the node. You can think of it as the return value in a function (i.e. `int Foo() { return 42; }`)
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name will be used as the data output slot's name | `Name="Return Value"` | 
+| __Type__ | Required | This is the C++ type of the property. It must be a type exposed to the O3DE serialization context. If the type contains special characters (e.g. `AZStd::vector<int>`) you must use the HTML Entity Codes (e.g. `AZStd::vector&lt;int&gt;`) | `Type="float"` |
+| __Description__ | Optional | Displayed as a tooltip when hovering over the slot | `Description="The radius of the circle"`|
+| __DefaultValue__ | Optional | Allows you to provide the default value for the property | `DefaultValue="1000.0"`|
+| __DisplayGroup__ | Optional | Allows slots to be grouped visually | `DisplayGroup="Tests"` |
+----
+
+### Examples
+
+
+### <a href="#branch"></a>Branch
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | Specifies the name of the branch, in C++ this will create a function with the name `Call<BranchName>` where `<BranchName>` is the specified name. 
+----
+
+### Examples
+
+The following example creates a node that will return `Success` and an `Entity Spawn Ticket` or return `Failed`
+```xml
+<Input Name="Create Ticket" OutputName="Ticket Created">
+    <Parameter Name="Prefab" Type="const AzFramework::Scripts::SpawnableScriptAssetRef&amp;" Description="Prefab source asset to spawn"/>
+	<Branch Name="Success">
+        <Return Name="Entity Spawn Ticket" Type="AzFramework::EntitySpawnTicket"/>
+    </Branch>
+    <Branch Name="Failed"/>
+</Input>
+```
+The above node's C++ code would be used in the following way:
+```cpp
+void CreateSpawnTicketNodeable::CreateTicket(const AzFramework::Scripts::SpawnableScriptAssetRef& prefab)
+{
+    auto ticket = m_spawnableScriptMediator.CreateSpawnTicket(prefab);
+    if (ticket.IsSuccess())
+    {
+        CallSuccess(ticket.GetValue());
+    }
+    else
+    {
+        AZLOG_ERROR("Unable to Create Spawn Ticket - A valid prefab was not provided");
```

Comment:
Is there a way for this log error to be generated automatically? If not, and this function was written manually, I'd remove the log error from this example.

Reply:
Wouldn't it be good to to provide best practices?

Reply:
This code is not autogen'd it's an example of how a user would write the inner-workings of a node. Does it read like this was autogen'd? I may need to add some clarification to avoid confusion

Reply:
Oh, I see that now. I think I misinterpreted the sentence, "The above node’s C++ code would be used in the following way" to mean that the XML in the example generated the code that follows. I'm not sure why I thought that. ;)

If I made that mistake, perhaps others might too, so I wonder if it would be helpful to either clarify a little more that the C++ code is calling into the code that the preceding XML generated, or perhaps include the C++ code that the XML generated in between the two. Thoughts?